src/workerd/server/server.c++

```diff
@@ -1535,14 +1660,52 @@ public:
       kj::Maybe<kj::StringPtr> entrypointName,
       kj::Maybe<kj::Own<Worker::Actor>> actor = kj::none) {
     TRACE_EVENT("workerd", "Server::WorkerService::startRequest()");
+
+    // Setting up tail workers support.
+
+    auto tracer = kj::rc<PipelineTracer>();
+    auto executionModel =
+        actor == kj::none ? ExecutionModel::STATELESS : ExecutionModel::DURABLE_OBJECT;
+    auto workerTracer =
+        tracer->makeWorkerTracer(PipelineLogLevel::FULL, executionModel, kj::none /* scriptId */,
+            kj::none /* stableId */, kj::none /* scriptName */, kj::none /* scriptVersion */,
+            kj::none /* dispatchNamespace */, nullptr /* scriptTags */, kj::none /* entrypoint */);
+
+    kj::Vector<kj::Own<WorkerInterface>> tailWorkers(loggingServices.size());
+    for (auto& svc: loggingServices) {
+      auto& service = lookupService(*svc, "looking logging service");
+      KJ_ASSERT(&service != this, "A worker currently cannot log to itself");
+      // Caution here... if the tail worker ends up have a cirular dependency
+      // on the worker we'll end up with an infinite loop trying to initialize.
+      // We can test this directly but it's more difficult to test indirect
+      // loops (dependency of dependency, etc). Here we're just going to keep
+      // it simple and just check the direct dependency.
+      tailWorkers.add(service.startRequest({}));
+    }
+
+    // When the tracer is complete, deliver the traces to both the parent
+    // and the tail workers.
+    waitUntilTasks.add(tracer->onComplete().then(
+        [tailWorkers = kj::mv(tailWorkers), parentTracer = mapAddRef(maybeTracer)](
```

Comment:
Missing kj::coCapture

Reply:
Not sure it's actually needed in this case but doesn't hurt anything.

Reply:
Well you're creating a task that will run a lot later and capturing variables that should still be alive after your first co_await so I think it's required